BigAnimal: CLI 2.0

drothery-edb opened the pull request:
## What Changed?

https://enterprisedb.atlassian.net/browse/UPM-18459

Updated command syntax but haven't verified all the example output yet - I was using these changes as an opportunity to verify the existing examples.

drothery-edb commented on product_docs/docs/biganimal/release/free_trial/detail/create_a_cluster/create_cluster_cli.mdx:
```diff
@@ -42,15 +47,15 @@ Switched the context credential to "newuser".
+│ name     │ address                     │ port │ context credential │
+├──────────┼─────────────────────────────┼──────┼────────────────────┤
+│ new-user │ portal.enterprisedb.network │ 443  │ x                  │
+└──────────┴─────────────────────────────┴──────┴────────────────────┘
```
Does the content of the following Caution need adjustments?

> If you add another credential, you’ll need to add `--credential [newuser]` to the following commands. If you have only one, the option isn't needed. You can change the default credential using `biganimal config set context_credential [name]`.

edbamolvyavahare replied:
If you add another credential, newly created credential will be set as default context_credential. you’ll need to add flag `--credential [newcreds]` to the commands to override the default credentials. If you have only one, the option isn't needed. You can change the default credential using `biganimal config set context_credential [name]`.

drothery-edb replied:
done

drothery-edb commented on product_docs/docs/biganimal/release/free_trial/detail/experiment/backup_and_restore.mdx:
````diff
@@ -100,13 +100,13 @@ For the free trial, you're limited to one active cluster. Restoring a cluster cr
 ```shell
-biganimal restore-cluster --id p-xxxxxxxxxxx --from-deleted --restore-point 2022-03-31T02:39:40+0000
+biganimal cluster restore --id p-xxxxxxxxxxx --from-deleted --restore-point 2022-03-31T02:39:40+0000
 __OUTPUT__
````
I wasn't able to verify this output because I wasn't able to connect to my cluster.

edbamolvyavahare commented on product_docs/docs/biganimal/release/free_trial/detail/create_a_cluster/create_cluster_cli.mdx:
````diff
@@ -24,15 +24,20 @@ chmod +x $HOME/bin/biganimal
 
 ```shell
-biganimal create-credential -a portal.biganimal.com -o 443 -n newuser
+biganimal credential create- -a portal.biganimal.com -o 443 -n newuser
````
```suggestion
biganimal credential create -n newuser
```

edbamolvyavahare replied:
-a and -o flag are for internal users

edbamolvyavahare commented on product_docs/docs/biganimal/release/free_trial/detail/create_a_cluster/create_cluster_cli.mdx:
````diff
@@ -98,27 +121,33 @@ highAvailability: false
+
+```
 __OUTPUT__
-Are you sure you want to create cluster "test_cluster"? [y|n]: y
````
We can keep this line, it still shows

drothery-edb commented on product_docs/docs/biganimal/release/reference/cli/managing_clusters.mdx:
````diff
@@ -174,27 +176,29 @@ After the cluster is created, you can update attributes of the cluster, includin
 
 ```shell
-biganimal show-clusters --name "my-biganimal-cluster" --provider "azure" \
+biganimal cluster show --name "my-biganimal-cluster" --provider "azure" \
````
I didn't validate the output from this point on to the end of the topic.

drothery-edb commented on product_docs/docs/biganimal/release/reference/cli/using_features.mdx:
```diff
@@ -7,13 +7,13 @@ navTitle: Using BigAnimal features
 
 You can use the faraway-replica-specific CLI commands to [create](#create-a-faraway-replica), [promote](#promote-a-faraway-replica), and [get information](#get-information-on-faraway-replicas) on faraway replicas. 
```
I didn't verify the output in the code blocks in this topic.

edbamolvyavahare commented on product_docs/docs/biganimal/release/free_trial/detail/experiment/backup_and_restore.mdx:
under
!!! Note Optional arguments for the CLI
We should also mention, if the interactive mode is disabled, then only those parameters will be updated which are provided as input from flags, values for all other parameters (which are not explicitly specified) will be inherited from the source cluster.

edbamolvyavahare commented on product_docs/docs/biganimal/release/getting_started/02_connecting_to_your_cloud/connecting_azure.mdx:
````diff
@@ -21,13 +21,13 @@ To connect your cloud:
 
    ```shell
-   biganimal create-credential --name <cred> --address portal.biganimal.com --port 443
+   biganimal credential create --name <cred> --address portal.biganimal.com --port 443
````
please remove mention of `--address portal.biganimal.com --port 443` flags, as these are for internal use

drothery-edb replied:
done

edbamolvyavahare commented on product_docs/docs/biganimal/release/getting_started/02_connecting_to_your_cloud/index.mdx:
````diff
@@ -47,27 +47,21 @@ For additional cloud provider-specific requirements, see [Setting up specific cl
 
    ```shell
-   biganimal create-credential --name <cred> --address portal.biganimal.com --port 443
+   biganimal credential create --name <cred> --address portal.biganimal.com --port 443
````
same as above

drothery-edb replied:
done

edbamolvyavahare commented on product_docs/docs/biganimal/release/getting_started/02_connecting_to_your_cloud/index.mdx:
````diff
@@ -47,27 +47,21 @@ For additional cloud provider-specific requirements, see [Setting up specific cl
 
    ```shell
-   biganimal create-credential --name <cred> --address portal.biganimal.com --port 443
+   biganimal credential create --name <cred> --address portal.biganimal.com --port 443
````
```suggestion
   biganimal credential create --name <cred>
```

edbamolvyavahare commented on product_docs/docs/biganimal/release/reference/cli/index.mdx:
```diff
@@ -50,10 +50,10 @@ chmod +x biganimal
-biganimal create-credential \
+biganimal credential create\
   --name "ba-user1" \
   --address "portal.biganimal.com" \
   --port "443"
```
need to remove mention of 
 --address "portal.biganimal.com" \
  --port "443"

drothery-edb replied:
done

edbamolvyavahare commented on product_docs/docs/biganimal/release/reference/cli/managing_clusters.mdx:
````diff
@@ -111,23 +112,23 @@ For backward compatibility, `allowIpRangeMap` and `pgConfigMap` properties also
+└────────┴───────────────────────────┴──────────┘
 ```
 
 !!!Note
````
`biganimal config set confirm_mode off`

edbamolvyavahare commented on product_docs/docs/biganimal/release/reference/cli/using_features.mdx:
```diff
@@ -7,13 +7,13 @@ navTitle: Using BigAnimal features
 You can use the faraway-replica-specific CLI commands to [create](#create-a-faraway-replica), [promote](#promote-a-faraway-replica), and [get information](#get-information-on-faraway-replicas) on faraway replicas. 
 
-To update, delete, and restore faraway replicas, use the `update-cluster`, `delete-cluster`, and `restore-cluster` commands. See [Managing clusters with the CLI](/biganimal/latest/reference/cli/managing_clusters/) for more information.
+To update, delete, and restore faraway replicas, use the `cluster update`, `cluster delete`, and `cluster restore` commands. See [Managing clusters with the CLI](/biganimal/latest/reference/cli/managing_clusters/) for more information.
```
command names needs to be updated 
faraway-replica create
faraway-replica update
faraway-replica delete

drothery-edb replied:
fixed

edbamolvyavahare commented on product_docs/docs/biganimal/release/reference/cli/using_features.mdx:
```diff
@@ -102,12 +102,12 @@ See [Other monitoring and logging solutions](/biganimal/latest/using_cluster/05_
 ## Maintenance windows CLI command
 
-You can set and view maintenance windows using the `cluster-maintenance-window` CLI command.
+You can set and view maintenance windows using the `cluster set-maintenance-window` CLI command.
```
You can set using: `cluster set-maintenance-window`
view using: `cluster get-maintenance-window`

drothery-edb commented:
@edbamolvyavahare Thanks for your review. I addressed all your comments. If you see anything else, let me know and I'll start another PR.

 @jasonparton if you see anything off, let me know. Same goes. I can always start another PR.
